### PR TITLE
feat: add unregister functions

### PR DESCRIPTION
**Pull request by yjhmelody**

### CODE REVIEW

File: ext.go [hunk 1]
```diff
@@ -34,20 +34,47 @@ func RegisterExt(extID int8, value MarshalerUnmarshaler) {
...
 ) {
 	typ := reflect.TypeOf(value)
 	extEncoder := makeExtEncoder(extID, typ, encoder)
+	oldTyp, ok := typeEncMap.Load(extID)
+	if ok {
+		typeEncMap.Delete(oldTyp)
+		typ := oldTyp.(reflect.Type)
+		if typ.Kind() == reflect.Ptr {
+			typeEncMap.Delete(typ.Elem())
+		}
```
Comment (vmihailenco) [inline ext.go]: This is rather subtle and repeats a lot.

I guess you are trying to fully unregister the old type - pointer to the type and the type. So
- if we pass the pointer - we unregister both
- if we pas the value - we unregister only the value and leave the pointer

That does not sound right...

Comment (yjhmelody) [reply ext.go:55]: I think unregister is the opposite behavior of register. What's your expected behavior?

Comment (yjhmelody) [reply ext.go]: Are there some errors in my code?

Comment (yjhmelody) [reply ext.go]: Or we call unregister func when we register?

Comment (vmihailenco) [reply ext.go]: >Or we call unregister func when we register?

And unexport `Unregister*` functions? Let's do this if it works for you.

Comment (vmihailenco) [reply ext.go]: Alternatively we can keep `UnregisterExt` but unexport `UnregisterExtEncoder` and `UnregisterExtDecoder`.

Comment (yjhmelody) [reply ext.go]: Ok, I think we should keep UnregisterExt at least.

**Comment by vmihailenco** [comment]
Thanks

**Pull request merged**